OMCompiler/Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -9404,7 +9411,51 @@ algorithm
   else
     variability := SimCodeVar.CONTINUOUS();
   end if;
-end setVariabilityAttribute;
+end getVariabilityAttribute;
+
+public function getDefaultInitialAttribute
+ "function which checks returns  defualt Initial attribute
+  value of a variable based on FMI-2.0 specification table for calculating
+  Initial attribute"
+  input SimCodeVar.Variability variability;
+  input SimCodeVar.Causality causality;
+  output SimCodeVar.Initial initial_;
+algorithm
+  initial_ := match(variability, causality)
+
+    case (SimCodeVar.CONSTANT(), SimCodeVar.OUTPUT()) then SimCodeVar.EXACT();
+    case (SimCodeVar.CONSTANT(), SimCodeVar.LOCAL()) then SimCodeVar.EXACT();
+
+    // variability=FIXED Vs causality = parameter, calculatedParameter and local
+    case (SimCodeVar.FIXED(), SimCodeVar.PARAMETER()) then SimCodeVar.EXACT();
+    case (SimCodeVar.FIXED(), SimCodeVar.CALCULATED_PARAMETER()) then SimCodeVar.CALCULATED();
+    case (SimCodeVar.FIXED(), SimCodeVar.LOCAL()) then SimCodeVar.CALCULATED();
+
+    // TODO varkind= tunable Vs causality = parameter, calculatedParameter and local, same like PARAM()
+
+    // variability=DISCRETE Vs causality = output and local
+    case (SimCodeVar.DISCRETE(), SimCodeVar.OUTPUT()) then SimCodeVar.CALCULATED();
+    case (SimCodeVar.DISCRETE(), SimCodeVar.LOCAL()) then SimCodeVar.CALCULATED();
+
+    // varkind=Continuous Vs causality = output and local
+    case (SimCodeVar.CONTINUOUS(), SimCodeVar.OUTPUT()) then SimCodeVar.CALCULATED();
+    case (SimCodeVar.CONTINUOUS(), SimCodeVar.LOCAL()) then SimCodeVar.CALCULATED();
+
+    else SimCodeVar.NONE_INITIAL();
+  end match;
+end getDefaultInitialAttribute;
+
+public function isDefaultVariability
+ "function which checks default causality attribute
+  causality = LOCAL() and return true or false "
+  input SimCodeVar.SimVar inVar;
+  output Boolean outBoolean;
+algorithm
+  outBoolean := match (inVar)
+    case (SimCodeVar.SIMVAR(variability = SOME(SimCodeVar.CONTINUOUS()))) then true;
+    else false;
```

Comment:
if `variability = NONE()` then the function should also return true.

Reply:
this function is not needed

Reply:
Not now - but it is probably the expected behavior. The current implementation could cause a potential bug in the future which might be difficult to catch.

Reply:
This is fixed with in more clear way using "function clearUpDefaultFmiAttributes" and the default values is set to NONE(), only after the function is called, so at this point, all attributes will have some values